mixin.js

```diff
@@ -351,12 +370,21 @@
         // loop through event group and trigger them
         for (i=0; i<event.length; i++)
         {
-          // detect key- events from touch- events
-          // TODO: Add support for mouse- events
-          eventObject = this[event[i].match(/^key/) ? '_createTypeEvent' : '_createTouchEvent'].apply(this, [event[i]].concat(extras));
-
-          // trigger event on target
-          target.trigger(eventObject);
+          switch(event) {
+            case 'focus':
```

Comment:
is there a way we can begin to make this stronger than magic strings? @alexindigo